waspc/src/Wasp/Generator/Common.hs

```diff
@@ -38,3 +43,36 @@ npmVersionRange =
 
 prismaVersion :: SV.Version
 prismaVersion = SV.Version 3 15 2
+
+npmCmd :: String
+npmCmd = case os of
+  -- Windows adds ".exe" to command, when calling it programmatically, if it doesn't
+  -- have an extension already, meaning that calling `npm` actually calls `npm.exe`.
+  -- However, there is no `npm.exe` on Windows, instead there is `npm` or `npm.cmd`, so we make sure here to call `npm.cmd`.
+  -- Extra info: https://stackoverflow.com/questions/43139364/createprocess-weird-behavior-with-files-without-extension .
+  "mingw32" -> "npm.cmd"
+  _ -> "npm"
+
+npxCmd :: String
+npxCmd = case os of
+  -- Read above, for "npm", why we need to handle Win in special way.
+  "mingw32" -> "npx.cmd"
+  _ -> "npx"
+
+buildNpmCmdWithArgs :: [String] -> (String, [String])
+buildNpmCmdWithArgs args = case os of
+  -- On Windows, due to how npm.cmd script is written, it happens that script
+  -- resolves some paths (work directory) incorrectly when called programmatically, sometimes.
+  -- Therefore, we call it via `cmd.exe`, which ensures this issue doesn't happen.
+  -- Extra info: https://stackoverflow.com/a/44820337 .
+  "mingw32" -> wrapCmdAndArgsInWinCmdExe (npmCmd, args)
+  _ -> (npmCmd, args)
+
+buildNpxCmdWithArgs :: [String] -> (String, [String])
+buildNpxCmdWithArgs args = case os of
+  -- Read above, for "npm", why we need to handle Win in special way.
+  "mingw32" -> wrapCmdAndArgsInWinCmdExe (npxCmd, args)
+  _ -> (npxCmd, args)
+
+wrapCmdAndArgsInWinCmdExe :: (String, [String]) -> (String, [String])
+wrapCmdAndArgsInWinCmdExe (cmd, args) = ("cmd.exe", [unwords $ "/c" : cmd : args])
```

Comment:
We should also consider escpaing the arguments here -> we are not doing any, and we might want to do some, to both make it more robust and reduce chance of injections.

Also, if we are using `cmd.exe`, from what I am getting, we might not have to provide it with `npm.cmd` but just `npm` might be enough, and it will know how to resolve that? Is that true? Investigate this.